src/runtime_src/xdp/profile/database/static_info_database.cpp

```diff
@@ -469,10 +481,11 @@ namespace xdp {
       return;
 
     // Check if new xclbin has new PL metadata
-    if (newXclbinType == XCLBIN_AIE_PL || newXclbinType == XCLBIN_PL_ONLY)
+    if ((newXclbinType == XCLBIN_AIE_PL) || (newXclbinType == XCLBIN_PL_ONLY))
     {
-      if (config->plDeviceIntf != nullptr)
-          delete config->plDeviceIntf; // It shouldn't be...
+      // Delete previous dummy plDeviceIntf if available and re-create using xclbin with PL
```

Comment:
If HW Ctx for new PL xclbin is created after a Dummy PL Device Interface (required only for AIE Trace) has been created, XDP (static DB) gives an error message and aborts the process. So, the "delete config->plDeviceIntf;" is unreachable. However, the comments here can be more explanatory.
```suggestion
      /* Delete previous dummy plDeviceIntf if available and re-create using xclbin with PL.
      * For now, if HW Ctx for PL xclbin is created after a dummy PL Device Interface has been created for AIE only xclbin (for aie_trace), 
      * XDP errors out and aborts while processing the new PL xclbin. 
      * So the following deletion of dummy PL Device Intf is unreachable.
      */
```